25 get schemas from request and response bodies

zeusttu opened the pull request:
Closes #25 and #20

Commit:
Add option to include request and response body schemas

Include schemas from:

paths/<path>/<httpmethod>/requestBody/content/<mimetype>/schema
paths/<path>/<httpmethod>/responses/<httpcode>/content/<mimetype>/schema

Closes #25

Commit:
Do not downcase filenames

Closes #20

Commit:
Move new functionality to separate function

adamkitain commented:
Love this PR -- any reason not to merge it? Was looking for exactly this.

adamkitain commented on openapi2jsonschema/util.py:
```diff
@@ -111,3 +113,49 @@ def append_no_duplicates(obj, key, value):
+        "application/vnd.api+json": "jsonapi",
+    }
+    result = {}
+    for mimetype, definition in body_definition["content"].items():
```
```suggestion
    for mimetype, definition in body_definition.get("content", {}).items():
```

adamkitain replied:
Responses don't necessarily have a content section.

zeusttu replied:
Applied the suggestion, thanks!

zeusttu committed:
Make "content" key in body definition optional

This was committed as a GitHub review suggestion.
Commit message suggested by Github:

Update openapi2jsonschema/util.py

Co-authored-by: Adam Kitain <adamkitain@gmail.com>

Ton-tesseract commented:
This PR contains a the fix for the blocker issue #20 and #40 
When will be merged? I am currently blocked because I don't own the swagger file so I cannot change entities names as a workaround
